### PR TITLE
Fixes for SDLNativeRules issues 

### PR DESCRIPTION
**Pull request by ericwol-msft**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `main` branch. 
  - [ ] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/src/iothub_client_core_ll.c [hunk 1]
```diff
@@ -833,6 +833,13 @@ static IOTHUB_CLIENT_CORE_LL_HANDLE_DATA* initialize_iothub_client(const IOTHUB_
                     config = client_config;
                 }
             }
+            else if (device_config == NULL)
```
Comment (ewertons) [inline iothub_client/src/iothub_client_core_ll.c:836]: Is this right?? on lines 775+ it looks like device_config == NULL is a valid scenario.

File: provisioning_client/tools/dice_device_enrollment/dice_device_enrollment.c [hunk 2]
```diff
@@ -17,7 +17,14 @@ static const char* const END_CERT_MARKER = "-----END CERTIFICATE-----";
 
 static char* get_user_input(const char* text_value, int max_len)
 {
-    char* result = malloc(max_len + 1);
+    int result_len = max_len + 1;
+    if (result_len <= 1)
+    {
+        (void)printf("invalid max_len size\r\n");
+        return NULL;
```
Comment (ewertons) [inline provisioning_client/tools/dice_device_enrollment/dice_device_enrollment.c:24]: It breaks our style of single point of return.

File: provisioning_client/tools/symm_key_provision/symm_key_provision.c [hunk 3]
```diff
@@ -20,7 +20,14 @@ typedef struct REGISTRATION_INFO_TAG
 
 static char* get_user_input(const char* text_value, int max_len)
 {
-    char* result = malloc(max_len + 1);
+    int result_len = max_len + 1;
+    if (result_len <= 1)
+    {
+        (void)printf("invalid max_len size\r\n");
+        return NULL;
```
Comment (ewertons) [inline provisioning_client/tools/symm_key_provision/symm_key_provision.c:27]: It breaks our style of single point of return.

File: iothub_client/src/iothub_client_core_ll.c (hunk 1 above)
Comment (ericwol-msft) [reply iothub_client/src/iothub_client_core_ll.c:836]: without it, it would fall into the else and at line 847 cause an AV

Comment (ericwol-msft) [reply iothub_client/src/iothub_client_core_ll.c:836]: fixed

File: provisioning_client/tools/dice_device_enrollment/dice_device_enrollment.c (hunk 2 above)
Comment (ericwol-msft) [reply provisioning_client/tools/dice_device_enrollment/dice_device_enrollment.c:24]: line 31 already does an early return

File: provisioning_client/tools/symm_key_provision/symm_key_provision.c (hunk 3 above)
Comment (ericwol-msft) [reply provisioning_client/tools/symm_key_provision/symm_key_provision.c:27]: line 34 already does an early return

**Review by ewertons — APPROVED** [review]

**Pull request merged**